### ISSUE TITLE
Error Playing Uploaded Files on Android 15

### ISSUE THREAD

**Issue by mebarbosa** [Automattic/pocket-casts-android#3346]
### Description

> It is downloaded and played locally. This happens with almost every file. Maybe one or two where it works. [...] Sometimes it stays at that stage and sometimes the whole system crashes and restarts.

9140979-zd-a8c

Another similar report: 9156237-zd-a8c

### Step-by-step reproduction instructions

We were not able to reproduce this issue in our end

### Screenshots or screen recording

See the video provided here: p1733831046713619/1733830216.036979-slack-C02A333D8LQ

### Did you search for existing bug reports?

- [x] I have searched for existing bug reports.

### Device, Operating system, and Pocket Casts app version

Android 15

**Comment by joashrajin** [comment]
User in 9156237-zd-a8c mentioned that it only occurs on episodes with artwork added

**Comment by ecronin** [comment]
Happening 100% of the time on my Pixel 6a since the December 5 Android update over the weekend.  Play updates are up to date at Nov 1, two applied since the crashes started but they haven't made a difference, it was on an Oct 1 update before applying the December system update so I don't think a Play update did it.

Can confirm that if I upload the exact same mp3, one with an embedded mjpeg 1920x1080 and one with just the audio stream, the one with a thumbnail puts System UI in a crash loop until some watchdog reboots the phone, the audio-only one plays fine.  If I adb in to the phone and remove the _imagefile associated with the mp3 with artwork pocketcasts no longer shows the artwork but it still causes the phone to crash when playing it (the one with no embedded artwork had no _imagefile)

from logcat around the time I tried playing the mp3 after deleting the _imagefile:
```
12-10 08:03:28.266  1952  1952 E AndroidRuntime: FATAL EXCEPTION: main
12-10 08:03:28.266  1952  1952 E AndroidRuntime: Process: com.android.systemui, PID: 1952
12-10 08:03:28.266  1952  1952 E AndroidRuntime: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no scheme was found for /stora...
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:185)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:166)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:814)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:2043)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1858)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.content.ContentResolver.openInputStream(ContentResolver.java:1528)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.graphics.ImageDecoder$ContentResolverSource.createImageDecoder(ImageDecoder.java:295)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.graphics.ImageDecoder.decodeBitmapImpl(ImageDecoder.java:1856)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at android.graphics.ImageDecoder.decodeBitmap(ImageDecoder.java:1848)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.graphics.ImageLoader.loadBitmapSync(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:19)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.graphics.ImageLoader$loadBitmap$2.invokeSuspend(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:45)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.graphics.ImageLoader$loadBitmap$2.invoke(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:13)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:5)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at kotlinx.coroutines.BuildersKt.withContext(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:74)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.graphics.ImageLoader.loadBitmap(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:15)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.loadBitmapFromUri(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:21)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.loadBitmapFromUriForUser(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:73)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.access$loadMediaDataForResumptionInBackground(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:221)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader$loadMediaDataForResumption$mediaData$1.invokeSuspend(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:47)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:8)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:110)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
12-10 08:03:28.266  1952  1952 E AndroidRuntime: 	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@6b396d8, Dispatchers.Main.immediate]
```

after that things get very angry with binder parcels to System UI not being delivered.  Same backtrace playing another mp3 with artwork and the _imagefile still present:

```
12-10 08:32:09.748  2040  2040 E AndroidRuntime: FATAL EXCEPTION: main
12-10 08:32:09.748  2040  2040 E AndroidRuntime: Process: com.android.systemui, PID: 2040
12-10 08:32:09.748  2040  2040 E AndroidRuntime: java.lang.IllegalArgumentException: Expected URL scheme 'http' or 'https' but no scheme was found for /stora...
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:185)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:166)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:814)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:2043)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:1858)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.content.ContentResolver.openInputStream(ContentResolver.java:1528)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.graphics.ImageDecoder$ContentResolverSource.createImageDecoder(ImageDecoder.java:295)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.graphics.ImageDecoder.decodeBitmapImpl(ImageDecoder.java:1856)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at android.graphics.ImageDecoder.decodeBitmap(ImageDecoder.java:1848)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.graphics.ImageLoader.loadBitmapSync(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:19)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.graphics.ImageLoader$loadBitmap$2.invokeSuspend(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:45)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.graphics.ImageLoader$loadBitmap$2.invoke(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:13)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:5)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at kotlinx.coroutines.BuildersKt.withContext(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:74)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.graphics.ImageLoader.loadBitmap(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:15)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.loadBitmapFromUri(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:21)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.loadBitmapFromUriForUser(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:73)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader.access$loadMediaDataForResumptionInBackground(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:221)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at com.android.systemui.media.controls.domain.pipeline.MediaDataLoader$loadMediaDataForResumption$mediaData$1.invokeSuspend(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:47)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:8)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at kotlinx.coroutines.DispatchedTask.run(go/retraceme 8c106884385aa0a1e9963b4857afcdde60c7039bc01f0eaa4caad696dd9f7e7a:110)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:307)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    at java.lang.Thread.run(Thread.java:1012)
12-10 08:32:09.748  2040  2040 E AndroidRuntime:    Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@445fc91, Dispatchers.Main.immediate]
```

**Comment by MiSikora** [comment]
I fixed it in #3351. Right now it's targeting `7.80` but we might to upstream it to older versions since it crashes OS.

**Comment by MiSikora** [comment]
We upstreamed the fix. It will be out with `7.78.2` and `7.79-rc-3`.
